src/asciidoc/docs/guides-general/build-custom-controls.adoc

```diff
@@ -0,0 +1,210 @@
+= Building custom controls
+:page-slug: build-custom-controls
+:page-description: How to build a custom control that captures user input.
+:includespath: ../_includes
+
+The Koji platform provides controls that you can use to capture user input, such as images, text, and sounds.
+In addition, you can create your own custom controls.
+
+Examples of custom controls are tile map editors, sound enhancers, or custom avatar creators, which enhance the app's interactivity.
+
+A custom control is an application in Koji; therefore, you create one the same way you create any other Koji application.
+However, unlike a typical Koji app, a custom control has only one context (Default).
+When triggered, it appears over the parent app like a modal.
+
+Because a custom control does not include other contexts, it cannot be used as a standalone app.
+
+== Creating a custom control
+
+=== Prerequisites
+
+include::{includespath}/prereqs.adoc[tags=webdev;remixproc]
+
+=== Steps
+
+. <<_create_a_new_project,Create a new project and install the required packages.>>
+. <<_code_your_custom_control,Code your custom control.>>
+. <<_provide_a_link_to_your_custom_control,Provide a link to your custom control.>>
+. <<_test_your_custom_control,Test your custom control.>>
+
+==== Create a new project
+
+. Fork the https://withkoji.com/@eltesorokoji/a5hv/details/[Custom Control Scaffold].
+
+include::{includespath}/steps-codeeditor.adoc[tag=rename-and-open-in-code-editor]
+
+include::{includespath}/steps-codeeditor.adoc[tag=cli-front-end]
+
+include::{includespath}/steps-codeeditor.adoc[tag=terminal-stop-process]
+
+. Install both @withkoji/core and @withkoji/custom-vcc-sdk packages.
++
+[source,bash]
+----
+npm install --save @withkoji/core
```

Comment:
Does the custom control app require the core package? It doesn't seem like it should.

Reply:
Removed. It was only needed by other things in the scaffold.

Reply:
There are still lines in the code sample that rely on the core package (the import statement on line 2 and the Koji.ready method call on lines 44 - 49). If the Koji.ready method is really needed, we’ll need to put the installation info back. If not, we’ll remove the extra lines from the code sample and scaffold.

Reply:
Restored the instructions to install @withkoji.core.